src/main/api/2/studio-api-2.yaml

```diff
@@ -5522,6 +5522,167 @@ paths:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                siteId: 
+                  type: string
+                  description: site ID
+                items:
+                  type: array
+                  description: path(s) of item(s)
+                  items:
+                    type: string
+                clearSystemProcessing:
+                  type: boolean
+                  description: true if system processing is to be cleared
+                clearUserLocked:
+                  type: boolean
+                  description: true if user lock is to be cleared
+                live:
+                  type: boolean
+                  description: true if item is to be set as live, otherwise false
+                staged:
+                  type: boolean
+                  description: true if item is to be set as staged, otherwise false
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+
+  /workflow/update_item_states:
+    post:
+      tags:
+        - workflow
+      summary: Set item workflow states for site using query
+      description: 'Required permission "set_item_states"'
+      operationId: updateItemStates
+      requestBody:
+        description: List of items to set workflow state(s)
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                siteId: 
+                  type: string
+                  description: site ID
+                paths:
+                  type: array
+                  description: path(s) regexes of item(s)
+                  items:
+                    type: string
+                states:
+                  type: integer
+                  format: int64
+                  description: state bitmap mask to filter by state
+                clearSystemProcessing:
+                  type: boolean
+                  description: true if system processing is to be cleared
+                clearUserLocked:
+                  type: boolean
+                  description: true if user lock is to be cleared
+                live:
+                  type: boolean
+                  description: true if item is to be set as live, otherwise false
+                staged:
+                  type: boolean
+                  description: true if item is to be set as staged, otherwise false
```

Comment:
- Where/what is the query?
- Weren't we going to have some sort of flag to apply to the entire query vs a given set of items?
- I'm wondering if these two mutating apis could be just one. They share much; we could send either query or item paths, and the backend would apply to either the entire query or the set of paths sent. All json properties - except for siteId - would be optional as they can be conjugated.

Reply:
@rart the query is the set of paths and bitmap at the beginning.

Reply:
@rart I don't think merging them is useful. They're quite different.